linked_list/linked_list.py

```diff
@@ -86,32 +143,80 @@ def visit(self):
 
     # method to reverse the singly linked list
     # note: the nodes should be moved and not just the values in the nodes
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
     def reverse(self):

```

Comment:
👍 Nice helper method, but the way you're doing this is causing this to become an O(n^2) function